flow/layers/child_scene_layer.cc

```diff
@@ -4,15 +4,42 @@
 
 #include "flow/layers/child_scene_layer.h"
 
+#include "mojo/skia/type_converters.h"
+
 namespace flow {
 
+// TODO(abarth): We need to figure out how to allocate these ids sensibly.
+static uint32_t next_id = 10;
+
 ChildSceneLayer::ChildSceneLayer() {
 }
 
 ChildSceneLayer::~ChildSceneLayer() {
 }
 
+void ChildSceneLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
+  transform_ = matrix;
+}
+
 void ChildSceneLayer::Paint(PaintContext::ScopedFrame& frame) {
 }
 
+void ChildSceneLayer::UpdateScene(mojo::gfx::composition::SceneUpdate* update,
+                                  mojo::gfx::composition::Node* container) {
+  uint32_t id = next_id++;
+
+  auto child_resource = mojo::gfx::composition::Resource::New();
```

Comment:
Ah, I see. And Paint() is a no-op for this class, so in other rasterisers it just doesn't do anything.

This is relatively subtle. The API looks like it's just a normal part of the API, and the dart:ui side of this is always exposed, so someone could easily come along and try to call UpdateScene thinking it's broken that the Dart-exposed API is a no-op. Can you add a comment in layer.h next to the UpdateScene abstract method declaration explaining UpdateScene's role?